src/lib/quoteEngine.js

```diff
@@ -26,25 +33,24 @@ const { formatEther } = ethers.utils;
  *
  * It sorts pools by the base price and then takes full capacity from the pools with
  * the lower prices until it reaches desired coverAmount
- * @param {number} coverAmount - The amount to be covered.
+ * @param {BigNumber} coverAmount - The amount to be covered.
  * @param {Array<object>} pools - An array of pool data objects.
  * @returns {object} - An object containing the allocations by pool ID. (poolId => BigNumber amount)
  */
-const calculateOptimalPoolAllocation = (coverAmount, pools) => {
-  // Pool Id (number) -> Capacity Amount (BigNumber)
+const calculatePoolAllocations = (coverAmount, pools) => {
+  // pool id (number) -> capacity amount
   const allocations = {};
-
   let coverAmountLeft = coverAmount;
 
-  pools.sort((a, b) => a.basePrice - b.basePrice);
-
   for (const pool of pools) {
-    let availableCapacity = pool.totalCapacity.sub(pool.initialCapacityUsed);
-
-    // don't take all available capacity because it can change when executing tx due to small change in NXM price
-    availableCapacity = availableCapacity.sub(availableCapacity.div(CAPACITY_MARGIN_DIVISOR));
-
-    if (availableCapacity.eq(0)) {
+    const actualCapacity = pool.totalCapacity.sub(pool.initialCapacityUsed);
```

Comment:
(nit) is `remainingCapacity` a more accurate term?

Reply:
also is it worth to add some debug logs if there is not enough allocations so that we can check what is going on?